### PR TITLE
feat: zrx support on bsc

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Adds zrx support for bsc

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

## Risk

Low

## Testing

- Ensure swaps work from bnb -> token, token -> token, and token -> bnb on bsc

### Engineering

:point_up: 

### Operations

:point_up: 

## Screenshots (if applicable)
image
image
image
image

### CODE REVIEW

File: src/components/Trade/hooks/useSwapper/utils.ts [hunk 1]
```diff
@@ -166,6 +167,13 @@ export const getDefaultAssetIdPairByChainId = (
             buyAssetId: 'eip155:10/erc20:0x4200000000000000000000000000000000000042',
           }
         : ethFoxPair
+    case KnownChainIds.BnbSmartChainMainnet:
+      return featureFlags.BnbSmartChainZrx
+        ? {
+            sellAssetId: bscAssetId,
+            buyAssetId: 'eip155:56/bep20:0xe9e7cea3dedca5984780bafc599bd69add087d56',
```
Comment (kaladinlight) [inline src/components/Trade/hooks/useSwapper/utils.ts:174]: BUSD

File: src/components/Trade/hooks/useSwapper/utils.ts [hunk 2]
```diff
@@ -166,6 +167,13 @@ export const getDefaultAssetIdPairByChainId = (
             buyAssetId: 'eip155:10/erc20:0x4200000000000000000000000000000000000042',
           }
         : ethFoxPair
+    case KnownChainIds.BnbSmartChainMainnet:
+      return featureFlags.BnbSmartChainZrx
```
Comment (gomesalexandre) [inline src/components/Trade/hooks/useSwapper/utils.ts:171]: Missing flag in `FeatureFlags`

Comment (kaladinlight) [reply src/components/Trade/hooks/useSwapper/utils.ts]: https://github.com/shapeshift/web/pull/3947/commits/ce778b42ae0937dc6a41b9dccae7ef573cc47401

**Review by gomesalexandre — COMMENTED** [review]
Retested according to the lib testing criteria:

### Preview

image
image

### Approval - confirmed the explorer link is correct

image

### Status step

image
image

Tx failed, see https://bscscan.com/tx/0xe6b64345d993e54405f46e2cf5afd4ae781864442ecbebf73ea642ede4c34ab6

Wondering if that could be because I somehow did a BNB-BUSD swap on Pancakeswap with that address previously?

**Comment by kaladinlight** [comment]
I did a test and was able to successfully swap: https://bscscan.com/tx/0xe8580b4406fbf60e50b54b0cb46b754563ed713cb4df932e6d4e97e5666d970c

File: .env.dev [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # feature flags
 REACT_APP_FEATURE_LIVESUPPORT=true
 REACT_APP_FEATURE_BNBSMARTCHAIN=true
-REACT_APP_FEATURE_DEFI_DASHBOARD=true
+REACT_APP_FEATURE_ZRX_BNBSMARTCHAIN=true
+REACT_APP_FEATURE_DEFI_DASHBOARD=false
```
Comment (0xdef1cafe) [inline .env.dev:5]:
```suggestion
REACT_APP_FEATURE_DEFI_DASHBOARD=true
```

File: src/components/Trade/hooks/useSwapper/useSwapper.tsx [hunk 4]
```diff
@@ -110,6 +110,10 @@ export const useSwapper = () => {
 
   const getTrade = useCallback(async () => {
     if (!wallet) throw new Error('no wallet available')
+    if (!sellAccountBip44Params) throw new Error('Missing sellAccountBip44Params')
```
Comment (0xdef1cafe) [inline src/components/Trade/hooks/useSwapper/useSwapper.tsx:113]: @0xApotheosis sane?

**Review by gomesalexandre — APPROVED** [review]
Tested again locally and confirmed https://github.com/shapeshift/web/pull/3947#issuecomment-1476560438 was an upstream issue:

### BUSD - BEP-20

image
image
image

### BEP-20 - BUSD

image
image
image

File: src/components/Trade/hooks/useSwapper/typeGuards.ts [hunk 5]
```diff
@@ -1,29 +1,18 @@
...
-): chainId is SupportedSwappingChain => {
-  return (
-    chainId === KnownChainIds.EthereumMainnet ||
-    chainId === KnownChainIds.AvalancheMainnet ||
-    chainId === KnownChainIds.OptimismMainnet ||
-    chainId === KnownChainIds.OsmosisMainnet ||
-    chainId === KnownChainIds.CosmosMainnet ||
-    chainId === KnownChainIds.ThorchainMainnet
-  )
+export const isEvmSwap = (chainId: ChainId): chainId is EvmChainId => {
```
Comment (0xApotheosis) [inline src/components/Trade/hooks/useSwapper/typeGuards.ts:10]: These names are so much better than what I used.

File: src/components/Trade/hooks/useSwapper/useSwapper.tsx (hunk 4 above)
Comment (0xApotheosis) [reply src/components/Trade/hooks/useSwapper/useSwapper.tsx:113]: Sane 👌

**Review by 0xApotheosis — APPROVED** [review]
Looks good to me. 

Ran locally as a sanity check - I get a lot of errors from `ENS.js` when using BSC, but I could also replicate that on Optimism on `develop`, so not an issue related to this PR.

File: .env.dev [hunk 6]
```diff
@@ -14,8 +14,8 @@ REACT_APP_UNCHAINED_AVALANCHE_HTTP_URL=https://dev-api.avalanche.shapeshift.com
 REACT_APP_UNCHAINED_AVALANCHE_WS_URL=wss://dev-api.avalanche.shapeshift.com
 REACT_APP_UNCHAINED_OPTIMISM_HTTP_URL=https://dev-api.optimism.shapeshift.com
 REACT_APP_UNCHAINED_OPTIMISM_WS_URL=wss://dev-api.optimism.shapeshift.com
-REACT_APP_UNCHAINED_BNBSMARTCHAIN_HTTP_URL=https://dev-api.bnbsmartchain.shapeshift.com
-REACT_APP_UNCHAINED_BNBSMARTCHAIN_WS_URL=wss://dev-api.bnbsmartchain.shapeshift.com
+REACT_APP_UNCHAINED_BNBSMARTCHAIN_HTTP_URL=https://api.bnbsmartchain.shapeshift.com
```
Comment (0xApotheosis) [inline .env.dev:17]: How temporary is this? Did we want to undo before merging this PR?

Comment (kaladinlight) [reply .env.dev:17]: nah, dev still borked and want to get some testing rolling. I will be sure to revert once stable again

**Pull request merged**